### PR TITLE
Formalize new package structure and cleanup

### PR DESCRIPTION
**Pull request by SUPERCILEX**
This PR reorganizes our package structure into something that will scale with the PRs coming in the future. It also does some relatively minor cleanup. (The large number of LoC changed are because of a single phone class.)

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
Here's what the new structure looks like visually:
- data
  - client
  - model
  - remote
- ui
  - email
  - idp
  - phone
  - provider (Future providers will go here. The existing ones do a bit of everything.)
- util
  - data
  - ui
    - fieldvalidators

File: auth/src/main/java/com/firebase/ui/auth/util/data/PhoneNumberUtils.java [hunk 1]
```diff
@@ -0,0 +1,658 @@
+package com.firebase.ui.auth.util.data;
```
Comment (SUPERCILEX) [inline auth/src/main/java/com/firebase/ui/auth/util/data/PhoneNumberUtils.java]: Here's the diff for this class: http://www.mergely.com/v0dlTATS/. Fixing it was _sooooo_ painful. I literally sat down with the ADB podcast and spent an hour simplifying it until I went insane and discovered this [unbelievably awesome IntelliJ feature](https://www.jetbrains.com/help/idea/structural-search-and-replace.html)—huge cudos to the JetBrains team. Anyway, how can I help you understand the diff?

Comment (SUPERCILEX) [reply auth/src/main/java/com/firebase/ui/auth/util/data/PhoneNumberUtils.java]: @samtstern Ok, I killed the `PhoneNumberUtils` changes and will move them to another PR so this diff can stay sane. 👍

**Comment by samtstern** [comment]
@SUPERCILEX we can look at this one next if you want to.  But let's keep it to renames only.  If we're making any code changes (even formatting changes) let's do those in another PR.

**Comment by SUPERCILEX** [comment]
@samtstern SGTM, done.

**Comment by samtstern** [comment]
LGTM.

**Pull request merged**